pkg/server/router/pipelineRecord.go

```diff
@@ -160,17 +161,39 @@ func (router *router) updatePipelineRecordStatus(request *restful.Request, respo
 		return
 	}
 
-	if e.PipelineRecord.Status == api.Running {
-		e.PipelineRecord.Status = api.Aborted
-		err = event.UpdateEvent(e)
-		if err != nil {
-			log.Errorf("update event %s error: %v", e.ID, err)
+	updatePipelineRecordStatusFromRunningToAborted(e.PipelineRecord)
+	// event and pipeline record both will be updated in method "UpdateEvent"
+	err = event.UpdateEvent(e)
+	if err != nil {
+		log.Errorf("update event %s error: %v", e.ID, err)
+	}
+
+	response.WriteHeaderAndEntity(http.StatusOK, pipelineRecord)
+}
+
+func updatePipelineRecordStatusFromRunningToAborted(p *api.PipelineRecord) {
```

Comment:
This method name is too looooong. What about `abortPipelineRecord`?

Reply:
ok